app/src/main/java/com/example/interntasks_2a/WeatherAdapter.kt

```diff
@@ -1,19 +1,26 @@
-package com.example.interntasks_2a.case1
+package com.example.interntasks_2a
 
-import android.content.Intent
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.annotation.DrawableRes
 import androidx.recyclerview.widget.RecyclerView
-import com.example.interntasks_2a.R
 import com.example.interntasks_2a.databinding.ItemWeatherBinding
 
 class WeatherAdapter(private val weatherList: List<Weather>) :
     RecyclerView.Adapter<WeatherAdapter.WeatherViewHolder>() {
 
-    override fun onCreateViewHolder(
-        parent: ViewGroup, viewType: Int
-    ): WeatherViewHolder {
+    interface OnItemClickListener {
```

Comment:
Bunu düzelt diye demiyorum , fakat high order function'ları araştırabilirsin. Bu yapılar biraz daha basittir. Görünce anlicaksın.

Reply:
Not aldım teşekkür ederim